libs/core/src/main/java/org/opensearch/core/util/ConfigurationUtils.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.core.util;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.OpenSearchParseException;
+
+import java.util.Map;
+
+/**
+ * Utility class for parsing configurations.
+ *
+ * @opensearch.internal
+ */
+public class ConfigurationUtils {
```

Comment:
Can we not reuse/extend the `Settings` specific classes

Reply:
i got the suggestion from @msfroh to reuse this `ConfigurationUtils` from ingest package, and I'm open to where to place it.
 @msfroh thoughts?

Reply:
Yeah... this class nicely handles any case where config is represented as a `Map<String, Object>`, which is reusable for ingest and search pipelines. Pipeline config is not able to be stored in `Settings`, since they may contain arrays of objects, which are not representable in the flat format that `Settings` use internally.

Once this PR is merged, I'm happy to modify search/ingest processors to use this class too.

Reply:
Oh -- minor nitpick related to this line: 

Since the constructor is already `private`, can we please mark this class `final`? (I feel like something usually nags me about that. Maybe there's an IntelliJ inspection?)

Reply:
Oh -- also, we should tag this one as `@opensearch.api` and stick the `@PublicAPI(since = "3.0.0")` annotation on it, since it's intended to be used by plugins, whether the `ingestion-kafka` plugin or various pipeline plugins.

Reply:
thanks for the suggestions